README.md

```diff
@@ -21,13 +21,63 @@ To build from source, you need to ensure that the headers and object files for t
 (cd c-dependencies/spidermonkey && sh download-engine.sh)
 ```
 
+
 Alternatively, the engine can also be built from source using `c-dependencies/spidermonkey/build-engine.sh`. That should only be required if you want to modify the engine itself, however.
 
+In additon you need to have the following tools installed to successfully build, and build from a linux based system.
+
+- Rust 
+  ```
+  curl -so rust.sh https://sh.rustup.rs && sh rust.sh -y
+  restart shell or run source $HOME/.cargo/env
+  ```
+- Build tools
+  ```sh
+  sudo apt install build-essential
+  ```
+- binaryen
+  ```sh
+  sudo apt install binaryen
+  ```
+- rust target wasm32-wasi
+  ```sh
+  rustup target add wasm32-wasi
+  ```
+- cbindgen
+  ```sh
+  cargo install cbindgen
+  ```
+- wasi-sdk
+  ```sh
+  curl -sS -L -O https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-12/wasi-sdk-12.0-linux.tar.gz
+  tar xf wasi-sdk-12.0-linux.tar.gz
+  sudo mkdir -p /opt/wasi-sdk
+  sudo mv wasi-sdk-12.0/* /opt/wasi-sdk/
```

Comment:
Is it worth pointing out that there's a `.deb` available for ubuntu/debian users?